modules/representation_management/lib/representation_management/v0/pdf_constructor/form_2122a.rb

```diff
@@ -0,0 +1,196 @@
+            # Claimant Email
+            "#{page1_key}.E_Mail_Address_Optional[0]": data.claimant_email,
+
+            # # Section III
+            # Representative Name
+            "#{page1_key}.Name_Of_Individual_Appointed_As_Representative_First_Name[0]": data.representative_first_name,
+            "#{page1_key}.Middle_Initial[0]": data.representative_middle_initial,
+            "#{page1_key}.Last_Name[0]": data.representative_last_name,
+            # Representative Type
+            "#{page1_key}.RadioButtonList[0]": representative_type_checkbox(data.representative_type)
+          }
+        end
+        # rubocop:enable Metrics/MethodLength
+
+        # rubocop:disable Layout/LineLength
+        def page2_options(data)
+          page2_key = 'form1[0].#subform[1]'
+          {
+            # Page 2
+            # Header Veteran SSN
+            "#{page2_key}.SocialSecurityNumber_FirstThreeNumbers[1]": data.veteran_social_security_number[0..2],
+            "#{page2_key}.SocialSecurityNumber_SecondTwoNumbers[1]": data.veteran_social_security_number[3..4],
+            "#{page2_key}.SocialSecurityNumber_LastFourNumbers[1]": data.veteran_social_security_number[5..8],
+            # Representative Mailing Address
+            "#{page2_key}.MailingAddress_NumberAndStreet[2]": data.representative_address_line1,
+            "#{page2_key}.MailingAddress_ApartmentOrUnitNumber[2]": data.representative_address_line2,
+            "#{page2_key}.MailingAddress_City[2]": data.representative_city,
+            "#{page2_key}.MailingAddress_StateOrProvince[2]": data.representative_state_code,
+            "#{page2_key}.MailingAddress_Country[2]": data.representative_country,
+            "#{page2_key}.MailingAddress_ZIPOrPostalCode_FirstFiveNumbers[2]": data.representative_zip_code,
+            "#{page2_key}.MailingAddress_ZIPOrPostalCode_LastFourNumbers[2]": data.representative_zip_code_suffix,
+            # Representative Phone Number
+            "#{page2_key}.Telephone_Number_Area_Code[2]": data.representative_phone[0..2],
+            "#{page2_key}.Telephone_Middle_Three_Numbers[1]": data.representative_phone[3..5],
+            "#{page2_key}.Telephone_Last_Four_Numbers[2]": data.representative_phone[6..9],
+            # Representative Email
+            "#{page2_key}.E_Mail_Address_Of_Individual_Appointed_As_Claimants_Representative_Optional[0]": data.representative_email_address,
+            # Record Consent
+            "#{page2_key}.AuthorizationForRepAccessToRecords[0]": data.record_consent == true ? 1 : 0,
+            # Consent Limits
+            "#{page2_key}.RelationshipToVeteran[1]": limitations_of_consent_text(data.consent_limits),
+            # Consent Address Change
+            "#{page2_key}.AuthorizationForRepActClaimantsBehalf[0]": data.consent_address_change == true ? 1 : 0
+          }
+        end
+        # rubocop:enable Layout/LineLength
+
+        def template_options(data)
```

Comment:
I like how you made #template_options shorter here. Please refactor `# page1_options` and `# page2_options` so you don't have to disable the `Layout/LineLength` cop.

Reply:
Btw, I think you can use your code comments to find suitable opportunities to to break methods like `#template_options` (in the other classes) `#page1_options`, etc. apart. For example you can have a method that does veteran fields, another that does claimant fields, etc.

Reply:
Good call, I split it up into several methods.